src/core/border_router/router_advertisement.cpp

```diff
@@ -110,9 +110,12 @@ void PrefixInfoOption::SetPrefix(const Ip6::Prefix &aPrefix)
     mPrefix       = static_cast<const Ip6::Address &>(aPrefix.mPrefix);
 }
 
-void PrefixInfoOption::GetPrefix(Ip6::Prefix &aPrefix) const
+Ip6::Prefix PrefixInfoOption::GetPrefix(void) const
 {
-    aPrefix.Set(mPrefix.GetBytes(), mPrefixLength);
+    Ip6::Prefix prefix;
+
+    prefix.Set(mPrefix.GetBytes(), mPrefixLength);
+    return prefix;
```

Comment:
Yes it is just for readability. Most compilers will do [NRVO](https://en.cppreference.com/w/cpp/language/copy_elision) to avoid unnecessary copies but it is not guaranteed. I am ok to revert it if we prefer performance against readability.

Reply:
I am OK with this as is. Thanks.